### PR TITLE
Use "selector" instead of "matcher"

### PR DESCRIPTION
**Pull request by glyn**
Fixes https://github.com/jsonpath-standard/internet-draft/issues/20

See the [rendered HTML](https://glyn.github.io/internet-draft/) version of this PR.

### CODE REVIEW

**Comment by glyn** [comment]
I particularly welcome @gregsdennis's comments since he raised https://github.com/jsonpath-standard/internet-draft/issues/20.

File: draft-normington-jsonpath-latest.txt [hunk 1]
```diff
@@ -191,98 +191,98 @@ Internet-Draft   JavaScript Object Notation (JSON) Path   September 2020
...
 
-   Each matcher acts on its input list of nodes as follows.  For each
-   node in the list, the matcher may or may not match the node.  If the
-   matcher matches the node, it selects zero or more descendants of the
-   node, or the node itself.  The output list of nodes of a matcher is
-   the concatenation of the lists of selected nodes for each input node.
+   Each selector acts on its input list of nodes as follows.  For each
+   node in the list, the selector selects zero or more nodes, each of
+   which is a descendant of the node or the node itself.  The output
+   list of nodes of a selector is the concatenation of the lists of
```
Comment (gregsdennis) [inline draft-normington-jsonpath-latest.txt:232]: "distinct concatenation"

**Review by gregsdennis — CHANGES_REQUESTED** [review]
I think the `matcher -> selector` changes are good.  I'd just add the word "distinct" in the place indicated.

Reading through, I saw a few other language nuances that could be cleared up.  I'll raise them in separate issues for discussion.

File: draft-normington-jsonpath-latest.txt (hunk 1 above)
Comment (glyn) [reply draft-normington-jsonpath-latest.txt:232]: Does distinct concatenation mean with duplicates removed? That term is unfamiliar to me.

Comment (gregsdennis) [reply draft-normington-jsonpath-latest.txt:232]: Correct, no duplicates. I think mine currently regards the same value from different locations as distinct from one another, meaning it'll keep both.

It's worth discussing further, probably. Maybe just merge this for now, and I'll open an issue for it.

**Review by gregsdennis — APPROVED** [review]

File: draft-normington-jsonpath-latest.txt (hunk 1 above)
Comment (glyn) [reply draft-normington-jsonpath-latest.txt:232]: Yes, I'd like to keep that as a separate issue. It's somewhat related to the placeholder we have in the spec about whether to remove duplicates from unions.

**Pull request merged**